src/cbmc/cbmc_parse_options.cpp

```diff
@@ -74,7 +75,7 @@ Function: cbmc_parse_optionst::cbmc_parse_optionst
 cbmc_parse_optionst::cbmc_parse_optionst(int argc, const char **argv):
   parse_options_baset(CBMC_OPTIONS, argc, argv),
   xml_interfacet(cmdline),
-  language_uit(cmdline, ui_message_handler),
+  language_uit(cmdline, ui_message_handler, std::make_shared<optionst>()),
```

Comment:
The policy is to always use the least appropriate means to solve a problem. I think this is not the case here. I don't see why shared ownership is required.

The problem that Reuben tries to solve here is to avoid duplicate parsing of options in language_Ui and *_parse_options, which is particularly annoying when default values are used when an option is not given. Therefore the idea is to use the options object also in language_Ui.

Yes, that's a major refactoring that requires a separate PR to introduce it.